CMakeLists.txt

```diff
@@ -2,47 +2,58 @@ cmake_minimum_required(VERSION 3.15)
 
 project(boost-asio-windows-sspi)
 
-set(Boost_USE_STATIC_LIBS ON)
-set(Boost_USE_MULTITHREADED ON)
-set(Boost_USE_STATIC_RUNTIME OFF)
-set(BOOST_ROOT "C:/boost_1_73_0/")
-set(BOOST_LIBRARYDIR "C:/boost_1_73_0/lib/")
+if (WIN32)
+  set(Boost_USE_STATIC_LIBS ON)
+  set(Boost_USE_MULTITHREADED ON)
+  set(Boost_USE_STATIC_RUNTIME OFF)
+  set(BOOST_ROOT "C:/boost_1_73_0/")
```

Comment:
umm.... :) probably best not to use a hard-coded directory. I believe `FindBoost` is the command, and it sets `$BOOST_ROOT` or something like that.

Reply:
Indeed it isn't, but in order to use `FindBoost` which is what the `FindPackage` command used just below does, it has to know the boost root directory.

I'm far from being an expert in CMake, but I believe it uses `pkg-config` on platforms where that is supported whereas you have to give it some place to look on Windows which is why I'm setting the root manually.

Anyway, it would most likely make sense to remove this specific line and just make the user have to specify that on the command line. Bare in mind that the user was only me and I didn't really plan for this file to stay :-)

Reply:
Yes, the normal thing in CMake is that if the user's boost is in a custom location (for example when cross-compiling) they would specify: `cmake -DBOOT_ROOT=<my_custom_directory> .` on the first invocation of CMake. The FindBoost.cmake script will know what to do from there.

https://cmake.org/cmake/help/latest/module/FindBoost.html